src/clj/rems/workflow/dynamic.clj

```diff
@@ -58,12 +58,19 @@
 (deftest test-all-event-types-handled
   (is (= EventTypes (set (get-event-types)))))
 
+(defmethod apply-event [:event/draft-saved :workflow/dynamic]
+  [application _workflow event]
+  (assoc application :draft-application {:items (:items event)
```

Comment:
application is so overloaded, how about just `:form` or `:form-contents`? do you need to mention the state in the key?